src/HotChocolate/Core/src/Execution/Processing/OperationContextPoolPolicy.cs

```diff
@@ -22,12 +21,20 @@ public bool Return(OperationContext obj)
                 return true;
             }
 
-            if (obj.Execution.TaskStats.IsCompleted)
+            // if work related to the operation context has completed or if it never started
+            // we can reuse the operation context.
+            if (obj.Execution.TaskStats.IsCompleted ||
+                (obj.Execution.TaskStats.AllTasks == 0 &&
+                    obj.Execution.TaskStats.NewTasks == 0 &&
+                    obj.Execution.TaskStats.CompletedTasks == 0))
             {
                 obj.Clean();
                 return true;
             }
 
+            // we also clean if we cannot reuse the context so that the context is
+            // gracefully discarded and can be garbage collected.
+            obj.Clean();
```

Comment:
we now in any case call clean here.

Reply:
I doubt. Does it safe to clean context if tasks have not completed yet (obj.Execution.TaskStats.IsCompleted = false and NewTasks != 0 in condition above)?

Reply:
In this case we are in an error case where the execution was aborted but there are still some tasks running that are waiting to be aborted. This happens in cases where users forget to honor the cancellation token. We do not want to reuse the context in this case since tasks could still be mutation the context. But, we need to actually clean it so that this can be garbage collected. In this case we are already in an error state and the tasks still running are running in error meaning if they fail because we clear the context should not really matter.

Reply:
Or is your concern that the clean could have an exception?

Reply:
I understand now. All is good. One note that Clean method set _isPooled = true, but it is not actually truth in that case (context will be garbaged at the end). However I believe it doesn't matter.

Reply:
I have renamed that to IsInitialized.